src/main/java/org/fenixedu/academic/domain/student/RegistrationDataByExecutionYear.java

```diff
@@ -67,7 +102,7 @@ public static RegistrationDataByExecutionYear getOrCreateRegistrationDataByYear(
                         .filter(registrationDataByExecutionYear -> registrationDataByExecutionYear.getExecutionYear() == executionYear)
                         .findAny();
 
-        return result.orElse(new RegistrationDataByExecutionYear(registration, executionYear));
+        return result.isPresent() ? result.get() : new RegistrationDataByExecutionYear(registration, executionYear);
```

Comment:
You may leave this as is, but a cleaner way of accomplishing this would be to use the `orElseGet` method of the `Optional` class.